cognite/client/data_classes/hosted_extractors/sources.py

```diff
@@ -284,7 +284,7 @@ def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None =
         elif type_ is None:
             raise KeyError("type is required")
         try:
-            return cast(Self, _MQTTAUTHENTICATION_WRITE_CLASS_BY_TYPE[type_]._load_authentication(resource))
+            return cast(Self, _AUTHENTICATION_WRITE_CLASS_BY_TYPE[type_]._load_authentication(resource))
         except KeyError:
             raise TypeError(f"Unknown authentication type: {type_}")
```

Comment:
This will give an incorrect error message if there is a `KeyError` raised in `_load_authentication`. (mistake in my original implementation)
```python
        try:
            authentication_cls = _AUTHENTICATION_WRITE_CLASS_BY_TYPE[type_]
        except KeyError:
            raise TypeError(f"Unknown authentication type: {type_}")
        return cast(Self, authentication_cls._load_authentication(resource))
```